### PR TITLE
Dataset and Fine-Tuning

### PR DESCRIPTION
**Pull request by giollimirgia**

### CODE REVIEW

File: src/worldcereal_cop4geoglam/finetuning_utils.py [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+    num_timesteps: int = 12,
+    timestep_freq: str = "month",
+    augment: bool = True,
+    time_explicit: bool = False,
+    task_type: Literal["binary", "multiclass"] = "binary",
+    num_outputs: int = 1,
+    classes_list: Optional[List[str]] = None,
+    # masking_strategy_train: MaskingStrategy = MaskingStrategy(MaskingMode.NONE),
+    # masking_strategy_val: MaskingStrategy = MaskingStrategy(MaskingMode.NONE),
+    label_jitter=0,
```
Comment (kvantricht) [inline src/worldcereal_cop4geoglam/finetuning_utils.py:43]: add type hinting for these two kwargs (sorry, it's also missing in the original one)

File: tests/worldcerealc4gtests/test_finetuning_utils.py
Comment (kvantricht) [inline tests/worldcerealc4gtests/test_finetuning_utils.py:1]: Make a quick new comparison to this one in worldcereal. I added a test which would have exposed the bug. This test may be a good one here as well. It adds a finetuning eval test for time-explicit and multiclass (a combo which was not yet tested).

File: src/worldcereal_cop4geoglam/datasets.py [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+                values = values[0]  # dem is not temporal
+                dem[..., DEM_BANDS.index(dst_atr)] = values
+            else:
+                raise ValueError(f"Unknown band {dst_atr}")
+        return dict(
+            s1=s1, s2=s2, meteo=meteo, dem=dem, latlon=latlon, timestamps=timestamps
+        )
+
+
+class Cop4GeoLabelledDataset(Cop4GeoDataset):
```
[the remark this answers is not in the document]
Comment (giollimirgia) [reply src/worldcereal_cop4geoglam/datasets.py:166]: I didn't find any difference except for the masking strategies being absent and the handling of `return_sample_id`

File: notebooks/data_preparation.ipynb
[the remark this answers is not in the document]
Comment (giollimirgia) [reply notebooks/data_preparation.ipynb:1]: This notebook was indeed more for me to implement the DBSCAN-based clustering. I will untrack the file and keep it locally. I will store the group-based splitting strategy here, in case at some point we'll want to test with that.

**Review by kvantricht — APPROVED** [review]
